hashes/src/sha256.rs

```diff
@@ -134,33 +146,55 @@ impl Hash {
     /// Computes hash from `bytes` in `const` context.
     ///
     /// Warning: this function is inefficient. It should be only used in `const` context.
-    pub const fn const_hash(bytes: &[u8]) -> Self { Hash(Midstate::const_hash(bytes, true).0) }
+    pub const fn const_hash(bytes: &[u8]) -> Self { Hash(Midstate::const_hash(bytes, true).bytes) }
 }
 
 /// Output of the SHA256 hash function.
```

Comment:
It's not a function, it's the `Midstate` struct.

Reply:
I've got no idea what that message means :) I created an issue because its not important for this PR. 

https://github.com/rust-bitcoin/rust-bitcoin/issues/3063

Reply:
oooo, I'm the goose, I thought we were talking about docs on the `const_hash` function :( I missed line 140/152

Reply:
Added a trailing patch, I don't actually know what finalizing means so I wrote that in the commit log. Is it just "check digest is not all zero" done in `sha256::Hash::from_engine` or is it a crypto concept?

Reply:
The not all zeroes is a fuzzing-related hack. Finalization is a crypto concept required for a hash to be secure.